### PR TITLE
AMD ROCm release 4.2: core components part-1

### PR DESCRIPTION
**Pull request by arjun-raj-kuppala**

### CODE REVIEW

File: var/spack/repos/builtin/packages/hsakmt-roct/package.py [hunk 1]
```diff
@@ -40,7 +42,7 @@ def install_targets(self):
...
 
     def cmake_args(self):
-        args = [
-            '-DBUILD_SHARED_LIBS="on"',
-        ]
-        return args
+        if '+shared' in self.spec:
+            return ['-DBUILD_SHARED_LIBS=ON']
+        else:
+            return ['-DDBUILD_SHARED_LIBS=OFF']
```
Comment (alalazo) [inline var/spack/repos/builtin/packages/hsakmt-roct/package.py]:
```suggestion
        return [
            self.define_from_variant('BUILD_SHARED_LIBS', 'shared')
        ]
```

File: var/spack/repos/builtin/packages/rocm-cmake/package.py [hunk 2]
```diff
@@ -27,8 +28,12 @@ class RocmCmake(CMakePackage):
...
+    variant('ldconfig', default=True, description='ROCm ldconfig')
 
     depends_on('cmake@3:', type='build')
 
     def cmake_args(self):
-        return ['-DROCM_DISABLE_LDCONFIG=ON']
+        if '+ldconfig' in self.spec:
+            return ['-DROCM_DISABLE_LDCONFIG=ON']
+        else:
+            return ['-DROCM_DISABLE_LDCONFIG=OFF']
```
Comment (alalazo) [inline var/spack/repos/builtin/packages/rocm-cmake/package.py]: Same as comment above

File: var/spack/repos/builtin/packages/rocm-smi-lib/package.py [hunk 3]
```diff
@@ -28,9 +29,16 @@ class RocmSmiLib(CMakePackage):
...
     variant('build_type', default='Release', values=("Release", "Debug"), description='CMake build type')
+    variant('shared', default=True, description='Build shared or static library')
 
     depends_on('cmake@3:', type='build')
 
+    def cmake_args(self):
+        if '+shared' in self.spec:
+            return ['-DBUILD_SHARED_LIBS=ON']
+        else:
+            return ['-DDBUILD_SHARED_LIBS=OFF']
```
Comment (alalazo) [inline var/spack/repos/builtin/packages/rocm-smi-lib/package.py]: Same as comment above

**Review by alalazo — CHANGES_REQUESTED** [review]
Minor requests, otherwise LGTM

**Review by haampie — APPROVED** [review]

**Comment by arjun-raj-kuppala** [comment]
@alalazo 
Can you help review and merge this change

**Comment by arjun-raj-kuppala** [comment]
@alalazo can you help merge this?

**Review by alalazo — APPROVED** [review]

**Pull request merged**